lib/ansible/modules/cloud/huawei/hwc_smn_topic.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 Huawei
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+###############################################################################
+# Documentation
+###############################################################################
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ["preview"],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: hwc_smn_topic
+description:
+    - Represents a SMN notification topic resource.
+short_description: Creates a resource of SMNTopic in Huaweicloud Cloud
+version_added: '2.8'
+author: Huawei Inc. (@huaweicloud)
+requirements:
+    - requests >= 2.18.4
+    - keystoneauth1 >= 3.6.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in Huaweicloud Cloud.
+        type: str
+        choices: ['present', 'absent']
+        default: 'present'
+    display_name:
+        description:
+            - Topic display name, which is presented as the name of the email
+              sender in an email message. The topic display name contains a
+              maximum of 192 bytes.
+        type: str
+        required: false
+    name:
+        description:
+            - Name of the topic to be created. The topic name is a string of 1
+              to 256 characters. It must contain upper- or lower-case letters,
+              digits, hyphens (-), and underscores C(_), and must start with a
+              letter or digit.
+        type: str
+        required: true
+extends_documentation_fragment: hwc
+'''
+
+EXAMPLES = '''
+- name: create a smn topic
+  hwc_smn_topic:
+      identity_endpoint: "{{ identity_endpoint }}"
+      user_name: "{{ user_name }}"
+      password: "{{ password }}"
+      domain_name: "{{ domain_name }}"
+      project_name: "{{ project_name }}"
+      region: "{{ region }}"
+      name: "ansible_smn_topic_test"
+      state: present
+'''
+
+RETURN = '''
+    create_time:
```

Comment:
the additioanl indentation is seems unnecessary, would you mind to streamline it?

Reply:
Thanks!  Done  and added integration tests.